android/src/main/java/com/idehub/GoogleAnalyticsBridge/GoogleAnalyticsBridge.java

```diff
@@ -280,6 +281,18 @@ public void setClient(String trackerId, String clientId)
         }
     }
 
+    @ReactMethod
+    public void getClientId(String trackerId, Callback successCallBack)
+    {
+        Tracker tracker = getTracker(trackerId);
+        String clientId = "";
+        if (tracker != null) {
+            clientId = tracker.get("&cid");
+        }
+
+        successCallBack.invoke(clientId);
```

Comment:
Rather use Promise-callback, see [here](https://github.com/idehub/react-native-google-analytics-bridge/blob/d90d15316f0df843fd2f3063569aec18824990de/android/src/main/java/com/idehub/GoogleAnalyticsBridge/GoogleAnalyticsBridge.java#L487) for example.